src/html5/tokenizer/state.rs

```diff
@@ -1,78 +1,78 @@
 /// These are the states in which the tokenizer can be in.
-#[derive(Debug, Copy, Clone)]
+#[derive(Debug, Copy, Clone, PartialEq)]
 pub enum State {
-    DataState,
-    CharacterReferenceInDataState,
-    RcDataState,
-    CharacterReferenceInRcDataState,
-    RawTextState,
-    ScriptDataState,
-    PlaintextState,
-    TagOpenState,
-    EndTagOpenState,
-    TagNameState,
-    RcDataLessThanSignState,
-    RcDataEndTagOpenState,
-    RcDataEndTagNameState,
-    RawTextLessThanSignState,
-    RawTextEndTagOpenState,
-    RawTextEndTagNameState,
-    ScriptDataLessThenSignState,
-    ScriptDataEndTagOpenState,
-    ScriptDataEndTagNameState,
-    ScriptDataEscapeStartState,
-    ScriptDataEscapeStartDashState,
-    ScriptDataEscapedState,
-    ScriptDataEscapedDashState,
-    ScriptDataEscapedDashDashState,
-    ScriptDataEscapedLessThanSignState,
-    ScriptDataEscapedEndTagOpenState,
-    ScriptDataEscapedEndTagNameState,
-    ScriptDataDoubleEscapeStartState,
-    ScriptDataDoubleEscapedState,
-    ScriptDataDoubleEscapedDashState,
-    ScriptDataDoubleEscapedDashDashState,
-    ScriptDataDoubleEscapedLessThanSignState,
-    ScriptDataDoubleEscapeEndState,
-    BeforeAttributeNameState,
-    AttributeNameState,
-    AfterAttributeNameState,
-    BeforeAttributeValueState,
-    AttributeValueDoubleQuotedState,
-    AttributeValueSingleQuotedState,
-    AttributeValueUnquotedState,
-    CharacterReferenceInAttributeValueState,
-    AfterAttributeValueQuotedState,
-    SelfClosingStartState,
-    BogusCommentState,
-    MarkupDeclarationOpenState,
-    CommentStartState,
-    CommentStartDashState,
-    CommentState,
-    CommentLessThanSignState,
-    CommentLessThanSignBangState,
-    CommentLessThanSignBangDashState,
-    CommentLessThanSignBangDashDashState,
-    CommentEndDashState,
-    CommentEndState,
-    CommentEndBangState,
-    DocTypeState,
-    BeforeDocTypeNameState,
-    DocTypeNameState,
-    AfterDocTypeNameState,
-    AfterDocTypePublicKeywordState,
-    BeforeDocTypePublicIdentifierState,
-    DocTypePublicIdentifierDoubleQuotedState,
-    DocTypePublicIdentifierSingleQuotedState,
-    AfterDoctypePublicIdentifierState,
-    BetweenDocTypePublicAndSystemIdentifiersState,
-    AfterDocTypeSystemKeywordState,
-    BeforeDocTypeSystemIdentifierState,
-    DocTypeSystemIdentifierDoubleQuotedState,
-    DocTypeSystemIdentifierSingleQuotedState,
-    AfterDocTypeSystemIdentifierState,
-    BogusDocTypeState,
-    CDataSectionState,
-    CDataSectionBracketState,
-    CDataSectionEndState,
+    AfterAttributeName,
```

Comment:
The reason this was not alphabetical is that it matches the states as defined in the specification (https://html.spec.whatwg.org).

I don't mind changing this to alphabetical though.

Reply:
Might just be me personally, but I think it will be easier to find the states if they're alphabetized.  (I was thinking of adding comments for the sections in a later PR.)